### PR TITLE
Fix incorrect .row-margin when using odd gutter size

### LINKED ISSUE #20106: Incorrect `margin-left` and `margin-right` on `.row`-class when using odd `@grid-gutter-width`
**Steps to reproduce**
1. Choose an odd `@grid-gutter-width`, e.g. `@grid-gutter-width: 15px`
2. Compile bootstrap Less-stylesheets

**Problem**
The left and right margin `.row` class, which should normally "cancel out" the left and right padding of the `.col-*-*` classes is actually flipped so that instead cancelling out, there's a one pixel difference between the two:
```css
.row {
  margin-left:  -7px; /* should be -8px */
  margin-right: -8px; /* should be -7px */
}

.col-md-12 {
  padding-left:  8px;
  padding-right: 7px;
}
```

**Reason**
The reason why this happens is that in `.row` the `ceil()` and `floor()`-functions [are used on negative numbers]: `margin-left: ceil(-7.5px) == -7px`, as opposed to the `.col-*-*`-classes where it's used in a standard way [on positive numbers]: `padding-left: ceil(7.5px) == 8px`, which ends up giving the one pixel difference. As `ceil()`/`floor()` rounds towards negative/positive infinity, rather than towards/away from zero.

[are used on negative numbers]: https://github.com/twbs/bootstrap/blob/7c86bc9664ddc3a1965fe91d7a0db97eb9348448/less/mixins/grid.less#L16-L17

[on positive numbers]: https://github.com/twbs/bootstrap/blob/7c86bc9664ddc3a1965fe91d7a0db97eb9348448/less/mixins/grid-framework.less#L22-L23

The bug was introduced with the fix for: https://github.com/twbs/bootstrap/issues/16281

### PR DESCRIPTION
**Pull request by malthejorgensen**
Fix #20106

### CODE REVIEW

File: less/mixins/grid.less [hunk 1]
```diff
@@ -13,8 +13,10 @@
 
 // Creates a wrapper for a series of columns
 .make-row(@gutter: @grid-gutter-width) {
-  margin-left:  ceil((@gutter / -2));
-  margin-right: floor((@gutter / -2));
+  @_margin_left_helpvar:  ceil((@gutter / 2));
```
Comment (cvrebert) [inline less/mixins/grid.less]: Don't bother using variables for these, just put them directly into the property values.

Comment (malthejorgensen) [inline less/mixins/grid.less]: I had to use the "helpvars" to get it to work 😞 
E.g. having `margin-left: -ceil((@gutter / 2));` in Less outputs `margin-left: -ceil(7.5px);` directly into the CSS (and not `-8px`).
I was unable to find a way of writing it that Less understood – without using the extra variables.

Comment (cvrebert) [inline less/mixins/grid.less]: Try adding another set of parentheses around the whole expression.

Comment (malthejorgensen) [inline less/mixins/grid.less]: Less's "Strict math" option is set to true in `Gruntfile.js`, which makes the behavior pretty weird (set "Strict math" to enabled in the options tab at the top of the page before trying the examples):
A. [One set of parenthesis](http://less2css.org/#%7B%22less%22%3A%22.row%20%7B%5Cn%20%20%40gutter%3A%2015px%3B%5Cn%20%20margin-left%3A%20%20-(ceil((%40gutter%20%2F%202)))%3B%5Cn%20%20margin-right%3A%20-(floor((%40gutter%20%2F%202)))%3B%5Cn%7D%22%7D) -- doesn't work
B. [Two sets of parenthesis](http://less2css.org/#%7B%22less%22%3A%22.row%20%7B%5Cn%20%20%40gutter%3A%2015px%3B%5Cn%20%20margin-left%3A%20%20-((ceil((%40gutter%20%2F%202))))%3B%5Cn%20%20margin-right%3A%20-((floor((%40gutter%20%2F%202))))%3B%5Cn%7D%22%7D) -- works

I'll change it to use double parenthesis around the `ceil()` and `floor()` functions

Comment (malthejorgensen) [inline less/mixins/grid.less]: Maybe that's what you meant – I think I understand what you were saying now 😝 
Well, it's updated now – double parenthesis and all, and it works without the "helpvars".

**Comment by mdo** [comment]
Closing per https://github.com/twbs/bootstrap/issues/20106#issuecomment-225404602.

**Pull request closed without merging**